### ISSUE TITLE
Formatting fixer doesn't fix formatting diagnostic

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#72534]
Considering the following sample:
```

    class C
    {
        void M()
        {
            string[] x = [
"""
string
""",
"""
string
"""];
        }
    }
```

A formatting diagnostic is produced following the comma inside the collection expression:
![image](https://github.com/dotnet/roslyn/assets/12466233/4092df55-9813-4b6d-b93a-aa415455804b)

When invoking the format fixer, nothing happens.
When previewing the formatting fix, an empty box is displayed:

![image](https://github.com/dotnet/roslyn/assets/12466233/b3eef170-2e6c-4423-a09d-80480cf0fbd6)

I still question the choice to produce that diagnostic, given that raw string literals are not prescriptive about whether to indent and some people (like myself) prefer to left indent them for clarity.  
But regardless of that design question, I would expect the fixer to be able to fix formatting issues that are reported.

**Comment by jcouv** [comment]
Here's another instance where:
1. I think the formatting diagnostic is questionable (ie. I think the formatting is fine as-is)
2. invoking the formatter fixer doesn't fix the formatter diagnostics that are display

![image](https://github.com/dotnet/roslyn/assets/12466233/98415533-3794-4461-8d11-1ad6217bbce4)

**Comment by CyrusNajmabadi** [comment]
Generally speaking, the formatter is opinionated (always has been) :).  Until we have an extensible system for tweaking rules by code, teh guidance is simply:

1. use suppressions to disable formatting rules in files where deviations are common.
2. do not turn on formatting warnings if that opinionated behavior clashes with the teams' desire for flexibility.

Turning the warnings is effectively stating "its better to be consistent with all of these cases, even if occasionally there are times when someone doesn't like how it looks.

--

Note: wrt to your case, i think this is because the opinion of hte formatter is that since `([` is on the same line, that only one indent should happen.  If you changed the above to:

```
Assert.Equal(
    [
        "i think the indentation here will be felt to be 2 in
    ],
    model....);
```

**Comment by jcouv** [comment]
Forcing the opening bracket to be a new line to get the indentation is worse than the formatting behavior of object creation; feels sad:

![image](https://github.com/dotnet/roslyn/assets/12466233/af385d88-d65f-40f2-ac26-4ccf6a70e9c1)

**Comment by jcouv** [comment]
Ran into another occurrence where the formatting fix doesn't fix the formatting squiggles (there's an extra space before the closing bracket):
![image](https://github.com/dotnet/roslyn/assets/12466233/c7567cd3-8d63-43da-a89e-cbbcf90bc74c)

**Comment by CyrusNajmabadi** [comment]
that def seems like a bug :)
